test/Toggle/MysqlMarketableToggleGatewayTest.php

```diff
@@ -0,0 +1,412 @@
+                'description_of_implementation_reason' => 'test',
+                'description_of_location' => 'of',
+                'guide_url' => 'marketing',
+                'app_notification_copy_text' => 'information'
+            ] );
+
+        $this->validateMarketingInformationInTheDatabase( $expected, $toggleId );
+    }
+
+    /**
+     * @test
+     */
+    public function givenExistantToggleIdWithAllMarketingInformation_whenAllInformationProvided_AllTheInformationWillBeUpdated()
+    {
+        $releaseId = $this->addRelease( 'Test Marketing Information for toggle 2', "Very Very useful" );
+        $toggleId = $this->addToggle( "MarketingToggleTest 2", $releaseId, true, "this blah...", "is blah...",
+            "the blah...", "test blah...", "of blah...",
+            "marketing blah...", "information blah..." );
+
+        $expected = [ [
+            'toggle_id' => $toggleId,
+            'screenshot_urls' => 'this',
+            'description_of_toggle' => 'is',
+            'description_of_functionality' => 'the',
+            'description_of_implementation_reason' => 'test',
+            'description_of_location' => 'of',
+            'guide_url' => 'marketing',
+            'app_notification_copy_text' => 'information'
+        ] ];
+        $this->gateway->setMarketingInformationForToggle( $toggleId,
+            [
+                'screenshot_urls' => 'this',
+                'description_of_toggle' => 'is',
+                'description_of_functionality' => 'the',
+                'description_of_implementation_reason' => 'test',
+                'description_of_location' => 'of',
+                'guide_url' => 'marketing',
+                'app_notification_copy_text' => 'information'
+            ] );
+
+        $this->validateMarketingInformationInTheDatabase( $expected, $toggleId );
+    }
+
+    /**
+     * @test
+     */
+    public function givenExistantToggleIdWithAllMarketingInformation_whenSomeInformationProvided_OnlyProvidedInformationWillBeUpdated()
+    {
+        $releaseId = $this->addRelease( 'Test Marketing Information for toggle 2', "Very Very useful" );
+        $toggleId = $this->addToggle( "MarketingToggleTest 2", $releaseId, true, "this blah...", "is blah...",
+            "the blah...", "test blah...", "of blah...",
+            "marketing blah...", "information blah..." );
+
+        $expected = [ [
+            'toggle_id' => $toggleId,
+            'screenshot_urls' => 'this blah...',
+            'description_of_toggle' => 'is',
+            'description_of_functionality' => 'the blah...',
+            'description_of_implementation_reason' => 'test',
+            'description_of_location' => 'of blah...',
+            'guide_url' => 'marketing',
+            'app_notification_copy_text' => 'information blah...'
+        ] ];
+        $this->gateway->setMarketingInformationForToggle( $toggleId,
+            [
+                'screenshot_urls' => '',
```

Comment:
I see lots of duplicate code with long arrays.  Can you clean it a little to make it more expressive of intent.